src/Emulator/Peripherals/Peripherals/Timers/ATCWDT200.cs

```diff
@@ -0,0 +1,410 @@
+//
+// Copyright (c) 2010-2023 Antmicro
+//
+// This file is licensed under the MIT License.
+// Full license text is available in 'licenses/MIT.txt'.
+//
+#define ATCWDT200_32BIT_TIMER
+using Antmicro.Renode.Core;
+using Antmicro.Renode.Peripherals.CPU;
+using Antmicro.Renode.Core.Structure.Registers;
+using Antmicro.Renode.Logging;
+using Antmicro.Renode.Time;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Antmicro.Renode.Peripherals.Miscellaneous;
+using System.Threading;
+
+namespace Antmicro.Renode.Peripherals.Timers
+{
+  public class ATCWDT200 : BasicDoubleWordPeripheral, IKnownSize
+  {
+    public ATCWDT200(Machine machine) : base(machine)
+    {
+      DefineRegisters();
+
+      IRQ0 = new GPIO();
+      IRQ1 = new GPIO();
+
+
+      interruptTimer = new LimitTimer(machine.ClockSource, timerFrequency, this, "interrupt_timer", InitialLimit, workMode: WorkMode.OneShot, eventEnabled: true);
+      interruptTimer.LimitReached += () =>
+      {
+        this.InfoLog("interrupt limit reached");
+        interruptPending.Value = true;
+
+        UpdateInterrupts();
+        resetTimer.Enabled = true;
+        this.InfoLog("enable watchdog reset timer {0}", resetTimer.Enabled);
+
+      };
+
+      resetTimer = new LimitTimer(machine.ClockSource, timerFrequency, this, "reset_timer", InitialLimit, workMode: WorkMode.OneShot, eventEnabled: true);
+      resetTimer.LimitReached += () =>
+           {
+             IRQ1.Set();
+             this.InfoLog("Sending  reset signal {0}", IRQ1.IsSet);
+
+             this.InfoLog("reset limit reached");
```

Comment:
We have to reduce the amount of logging. For sure, it is pretty useful during the development, but we need to decide which ones we really want to keep as they actually are useful